### ISSUE TITLE
2.2.0-M2 error: org.neo4j.graphdb.TransactionFailureException

### ISSUE THREAD

**Issue by cdshines** [neo4j/neo4j#3846]
After update from 2.1.6 to 2.2.0-M2 this exception started to show up (I didn't change anything else in the code):

     org.neo4j.graphdb.TransactionFailureException: Failure to begin transaction 
        at org.neo4j.kernel.InternalAbstractGraphDatabase.beginTx(InternalAbstractGraphDatabase.java:919)
        at org.neo4j.kernel.ha.HighlyAvailableGraphDatabase.beginTx(HighlyAvailableGraphDatabase.java:240)
    ...
    Caused by: org.neo4j.kernel.api.exceptions.TransactionFailureException: Kernel has encountered some problem, please perform neccesary action (tx recovery/restart)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
        at org.neo4j.kernel.KernelHealth.assertHealthy(KernelHealth.java:63)
        at org.neo4j.kernel.impl.api.Kernel.newTransaction(Kernel.java:82)
        at org.neo4j.kernel.InternalAbstractGraphDatabase.beginTx(InternalAbstractGraphDatabase.java:912)
    Caused by: java.lang.AbstractMethodError: org.neo4j.collections.indexprovider.TimelineIndexProvider$TimelineIndexImplementation.force()V
        at org.neo4j.kernel.impl.transaction.log.LogRotationControl.forceEverything(LogRotationControl.java:60)
        at org.neo4j.kernel.LogRotationImpl.doRotate(LogRotationImpl.java:94)
        at org.neo4j.kernel.LogRotationImpl.rotateLogIfNeeded(LogRotationImpl.java:60)
        at org.neo4j.kernel.impl.transaction.log.AbstractPhysicalTransactionAppender.append(AbstractPhysicalTransactionAppender.java:101)
        at org.neo4j.kernel.impl.transaction.log.BatchingPhysicalTransactionAppender.append(BatchingPhysicalTransactionAppender.java:37)
        at org.neo4j.kernel.impl.api.TransactionRepresentationCommitProcess.commitTransaction(TransactionRepresentationCommitProcess.java:82)
        at org.neo4j.kernel.impl.api.TransactionRepresentationCommitProcess.commit(TransactionRepresentationCommitProcess.java:52)
        at org.neo4j.kernel.ha.MasterTransactionCommitProcess.commit(MasterTransactionCommitProcess.java:53)
        at sun.reflect.GeneratedMethodAccessor336.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.neo4j.kernel.ha.DelegateInvocationHandler.proxyInvoke(DelegateInvocationHandler.java:107)
        at org.neo4j.kernel.ha.DelegateInvocationHandler.invoke(DelegateInvocationHandler.java:99)
        at com.sun.proxy.$Proxy90.commit(Unknown Source)
        at org.neo4j.kernel.impl.api.KernelTransactionImplementation.commit(KernelTransactionImplementation.java:478)
        at org.neo4j.kernel.impl.api.KernelTransactionImplementation.close(KernelTransactionImplementation.java:407)
        at org.neo4j.kernel.TopLevelTransaction.close(TopLevelTransaction.java:112)

I would also be glad to provide any further help/information in case you need anything.

**Comment by jexp** [comment]
Can you share more of your code?

**Comment by cdshines** [comment]
Yes, the place where it fails looks like this:

public boolean isTreePresent(final String key, final Object value) {
		Objects.requireNonNull(key);
		Objects.requireNonNull(value);

		try (final Transaction tx = db.beginTx()) {
			boolean present = db.findNodesByLabelAndProperty(Labels.ROOT, key, value).iterator().hasNext();
			tx.success();
			return present;
		}
	}

And it's called without any transaction nesting etc.

**Comment by jexp** [comment]
Thanks a lot, the code looks normal. Except for not closing the iterator returned.
Would you be able to create a project with a failing unit test?

**Comment by cdshines** [comment]
Yes, I'll try to reproduce this and share by the end of the day. About iterator - oops, I thought they are closed upon transaction's `close` call, but now I can't find any reference to this in the docs. Thanks.

**Comment by cdshines** [comment]
@jexp, I managed to resolve this issue. Our project had an old dependency on `graph-collections` and it looks like some kind of a conflict happened, due to which the `java.lang.AbstractMethodError` was thrown. 
The problem was running tests in IDE didn't help to reproduce the error, which became reproducible only when running a packaged application in docker.
I dropped that dependency and everything is alright now. Sorry for the trouble.
